application/src/main/java/org/togetherjava/tjbot/features/Features.java

```diff
@@ -161,6 +165,11 @@ public static Collection<Feature> createFeatures(JDA jda, Database database, Con
         features.add(new JShellCommand(jshellEval));
 
         FeatureBlacklist<Class<?>> blacklist = blacklistConfig.normal();
-        return features.stream().filter(f -> blacklist.isEnabled(f.getClass())).toList();
+        return features.stream().filter(f -> !blacklist.isEnabled(f.getClass())).peek(f -> {
+            // Log INFO level message for each disabled feature using the provided logger
+            String message = "Feature '" + f.getClass().getSimpleName() + "' is disabled.";
+            logger.info(message);
+        }).toList();
```

Comment:
as u can see using peek is forbidden, so just put the log in filter method.

Also ur changing the logic before it removed the blacklisted features and kept all others, now it'll do the opposite